WordPress/gutenberg
# Add aspect ratio presets support via theme.json

## What?
This PR adds support for aspect ratio presets via theme.json. It adds by default the same presets that are available for the crop utilities in the image block.

## Why?
Part of https://github.com/WordPress/gutenberg/issues/38990

## How?
Bootstraps the presets via the theme.json class.

## Testing Instructions
- Check out this PR and verify you see the presets in the following places:
    - Core Image Block Inspector Controls
    - Core Image Block Croping tools (Toolbar)
    - Core Featured Image Inspector Controls
    - Core Cover block Inspector Controls
- Add your own presets via the theme (`settings.dimensions.aspectRatios`) and verify they appear
- Disable the default presets `settings.dimensions.defaultAspectRatios` -> `false` and verify they do not appear

<details>
<summary> Sample `theme.json`</summary>

```json
{
	"version": 3,
	"settings": {
		"dimensions": {
			"defaultAspectRatios": true,
			"aspectRatios": [
				{
					"name": "Extra Wide - 2:1",
					"ratio": "2/1",
					"slug": "2-1"
				}
			]
		}
	}
}

```

</details>

### Testing Instructions for Keyboard

N/A, backend only change. 

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/20684594/9050f661-1452-4699-8822-93b2fd52466c
